### PR TITLE
#206 Fix problem with validation check on progress bar

### LINKED ISSUE #206: Bug: Fix problem with allowing user to go to another page (after the first one is valid) on linear Application

### PR DESCRIPTION
**Pull request by nmadruga**
Fix #206

### Changes
- Simply check for the previous page of the **aimed** page the user wants to go. Using the same `validatePreviousPage` in both `NavigationBox` and `ProgressBar` for reusability, although it made the process a bit less clear when click **Next**. 

### Steps to reproduce bug (on `master`):
1. Create a new "Review Test form" application
2. Fill all elements in page 1, loose focus on last on so it will validate
3. Now click on Section 2 **Product Info**, and it will wrongly take you there!

### CODE REVIEW

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Looks good, but just to clarify what we just discussed IRL:

- The name `validatePreviousPage` is confusing as it's not clear if you're validating the page previous to the one you're on, or the one your trying to go to.
- Suggest breaking into two functions:
   - `validatePage` which validates any page you pass to it (with a default to the current, perhaps?)
   - a function to get the Section and Page of the one you wish to validate (previous to the one you're trying to link to, isn't it?) which you then pass to `validatePage`.

Make sense?

File: src/containers/Application/ApplicationPageWrapper.tsx [hunk 1]
```diff
@@ -121,17 +120,33 @@ const ApplicationPageWrapper: React.FC = () => {
     setPageElements(elements)
   }, [responsesLoading, currentSection, page, elementsState])
 
-  const validateCurrentPage = (): boolean => {
-    if (!application || !currentSection || !page) {
-      console.log('Problem to validate - Undefined parameters')
+  const validatePreviousPage = (sectionCode: string, pageNumber: number): boolean => {
```
[the remark this answers is not in the document]
Comment (nmadruga) [reply src/containers/Application/ApplicationPageWrapper.tsx:123]: I'll try to work on this change now

File: src/components/Application/ProgressBar.tsx [hunk 2]
```diff
@@ -32,6 +39,11 @@ const ProgressBar: React.FC<ProgressBarProps> = ({
     return status ? indicator[status] : null
   }
 
+  const getPreviousPageIsValid = (sectionCode: string, pageNumber: number) => {
```
Comment (CarlosNZ) [inline src/components/Application/ProgressBar.tsx:42]: I kind of like functions that return booleans to start with `is` -- so `isPreviousPageValid`. 

No biggie though, I just think it reads clearer (as opposed to `get` which suggests I'm expecting a spefiic value back).

**Comment by nmadruga** [comment]
Forgot to say here - I have done the changes as suggested.

**Review by CarlosNZ — APPROVED** [review]
LGTM! 
Just the one tiny comment -- not essential.

File: src/components/Application/ProgressBar.tsx (hunk 2 above)
Comment (nmadruga) [reply src/components/Application/ProgressBar.tsx:42]: Oh, I agree. That's a good idea.
Will change this now and start merging PRs!

**Pull request merged**